File: verify_auth.js
// verify_auth.js checks auth flow using global fetch (Node 18+)

const BASE_URL = 'http://localhost:3001/api';

async function verifyAuth() {
    try {
        console.log('--- Verifying Authentication ---');

        // 1. Register
        const username = `user_${Date.now()}`;
        const password = 'password123';
        console.log(`Registering user: ${username}`);
        const regRes = await fetch(`${BASE_URL}/auth/register`, {
            method: 'POST',
            headers: { 'Content-Type': 'application/json' },
            body: JSON.stringify({ username, password, role: 'admin' })
        });
        const regData = await regRes.json();
        if (!regData.success) throw new Error('Registration failed: ' + regData.error);
        console.log('Registration successful');

        // 2. Login
        console.log('Logging in...');
        const loginRes = await fetch(`${BASE_URL}/auth/login`, {
            method: 'POST',
            headers: { 'Content-Type': 'application/json' },
            body: JSON.stringify({ username, password })
        });
        const loginData = await loginRes.json();
        if (!loginData.token) throw new Error('Login failed: ' + loginData.error);
        const token = loginData.token;
        console.log('Login successful, token received');

        // 3. Access Protected Route (Employees)
        console.log('Accessing protected route (Employees)...');
        const empRes = await fetch(`${BASE_URL}/employees`, {
            headers: { 'Authorization': `Bearer ${token}` }
        });
        if (empRes.status !== 200) throw new Error('Access failed: ' + empRes.status);
        console.log('Protected route access successful');

        // 4. Access without token (should fail)
        console.log('Accessing without token...');
        const failRes = await fetch(`${BASE_URL}/employees`);
        if (failRes.status !== 401 && failRes.status !== 403) throw new Error('Security check failed! Status: ' + failRes.status);
        console.log('Security check passed (access denied without token)');

        console.log('--- Verification Complete: SUCCESS ---');

    } catch (e) {
        console.error('--- Verification FAILED ---');
        console.error(e.message);
    }
}

verifyAuth();

File: verify_auth.py
import requests
import sys

BASE_URL = 'http://localhost:8000/api'
AUTH_URL = f"{BASE_URL}/auth/login"

def verify_auth():
    print(f"Logging in to {AUTH_URL}...")
    try:
        resp = requests.post(AUTH_URL, json={'username': 'admin', 'password': 'admin123'})
        print(f"Login Status: {resp.status_code}")
        if resp.status_code != 200:
            print("Login failed:", resp.text)
            return

        data = resp.json()
        token = data.get('token')
        print(f"Token received: {token[:20]}...")

    # Test valid request
        headers = {'Authorization': f'Bearer {token}'}
        print("\nTesting GET /projects/ with token...")
        r = requests.get(f"{BASE_URL}/projects/", headers=headers)
        print(f"Projects Status: {r.status_code}")
        if r.status_code == 200:
            print("SUCCESS: Token is valid for GET.")
        else:
            print(f"FAILED GET: {r.text}")

        # Test POST
        print("\nTesting POST /projects/ with token...")
        data = {
            'name': 'Auth Test Project',
            'client': 'Auth Client',
            'income': 500
        }
        r = requests.post(f"{BASE_URL}/projects/", json=data, headers=headers)
        print(f"POST Status: {r.status_code}")
        if r.status_code == 201:
             print("SUCCESS: Token is valid for POST.")
        else:
             print(f"FAILED POST: {r.text}")

    except Exception as e:
        print(f"Exception: {e}")

if __name__ == "__main__":
    verify_auth()
